4_gap_analysis/src/Con_byState.R

```diff
@@ -0,0 +1,162 @@
+  mutate(Year = as.integer(format(ActivityStartDateTime, '%Y'))) %>%
+  group_by(MonitoringLocationIdentifier, ActivityStartDateTime, Year) %>%
+  summarize(NumConstituentsPerSample=length(unique(ConstituentGroup))) %>%
+  ungroup()
+
+# count the number of samples per year, separating into samples with one
+# constituent or two
+samples_per_year <- constituents_per_sample %>%
+  group_by(MonitoringLocationIdentifier, Year) %>%
+  summarize(
+    NumSamplesPerYear_Min1Const=length(ActivityStartDateTime[NumConstituentsPerSample >= 1]),
+    NumSamplesPerYear_Min2Const=length(ActivityStartDateTime[NumConstituentsPerSample >= 2]),
+    NumSamplesPerYear_Min3Const=length(ActivityStartDateTime[NumConstituentsPerSample >= 3])) %>%
+  ungroup() %>%
+  gather(temporary, NumSamplesPerYear, starts_with('NumSamplesPerYear')) %>%
+  extract(temporary, 'MinNumConstituents', "NumSamplesPerYear_Min([[:alnum:]])Const") %>%
+  mutate(MinNumConstituents=as.integer(MinNumConstituents))
+
+# compute sampling frequencies for samples containing at least 1, at least 2, or
+# 3 different constitutent types
+sample_rates <- samples_per_year %>%
+  # make sure all 6 years are represented for each site, even if there were 0
+  # samples some years
+  filter(Year %in% 2010:2015) %>%
+  complete(Year = 2010:2015) %>%
+  mutate(NumSamplesPerYear = ifelse(is.na(NumSamplesPerYear), 0, NumSamplesPerYear)) %>%
+  # compute various stats for frequency over all target years
+  group_by(MonitoringLocationIdentifier, MinNumConstituents) %>%
+  summarize(
+    MeanSamplesPerYear=mean(NumSamplesPerYear),
+    MinSamplesPerYear=min(NumSamplesPerYear),
+    MaxSamplesPerYear=max(NumSamplesPerYear),
+    MedianSamplesPerYear=median(NumSamplesPerYear))
+
+# split into scenarios
+# Do some more work here
+scenario_empty <- sample_rates %>%
+  filter(MinNumConstituents == 3, MinSamplesPerYear >= 12)
+scenario_A <- sample_rates %>%
+  filter(MinNumConstituents == 2, MinSamplesPerYear >= 12)
+scenario_B <- sample_rates %>%
+  filter(MinNumConstituents == 1, MinSamplesPerYear >= 12)
+scenario_C <- sample_rates %>%
+  filter(MinNumConstituents == 2, MinSamplesPerYear >= 6)
+
+siteInfo<-rbind(scenario_A,scenario_B,scenario_C)
+siteInfo$cat<-ifelse(siteInfo$MonitoringLocationIdentifier %in% scenario_A$MonitoringLocationIdentifier,"A) 2 cons., >=12 samples/year",
+                     ifelse(siteInfo$MonitoringLocationIdentifier %in% scenario_B$MonitoringLocationIdentifier,"B) 1 cons., >=12 samples/year",
+                            ifelse(siteInfo$MonitoringLocationIdentifier %in% scenario_C$MonitoringLocationIdentifier,"C) 2 cons. >=6 samples/year","D) <6 samples/year")))
+
+state<-gsub("_"," ",state)
+stateAb<-state.abb[grep(state, state.name)]
+
+write_feather(siteInfo,paste("andy/",stateAb,"_samples.feather",sep=""))
+write.csv(siteInfo,paste("andy/",stateAb,"_samples.csv",sep=""),row.names=F,quote=F)
```

Comment:
Consider wrapping all of the above data prep code into a single function. It could be named something like `combine_interpret_solutes_1state()`, `combine_solutes`, or `harmonize_solutes`. The main benefits would come in having this function (1) acquire all of the information it needs as function arguments (including filenames) - this way all you (or we) need to do is look at the function declaration to know what the function depends on and which external changes would require this function to be rerun; and (2) distinguish the fully automated data processing code from the more interactive data exploration that follows; this would pave the way for integrating this file into our larger data pipeline if you wanted, and IMO is helpful for code interpretation and modifications by others, especially down the road when we've all forgotten the code details and need to relearn them.

Also, if you prefer .csv to .feather here, I have no objections to only writing the one .csv file and skipping the `write_feather()` call. This is also the location where I suggested that if you wanted an automated push to google drive, you could follow the pattern here: https://github.com/USGS-R/nawqa_wqp/blob/0d82423a201b5eb6e196e385e83fe74a617a183e/1_wqpdata/src/wqp_inventory.R#L81-L84